packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -427,6 +432,23 @@ class DevFS {
     });
     if (dirtyEntries.isNotEmpty) {
       printTrace('Updating files');
+      if (generator != null) {
+        final List<String> invalidatedFiles = <String>[];
+        dirtyEntries.forEach((Uri deviceUri, DevFSContent content) {
+          if (content is DevFSFileContent) {
+            invalidatedFiles.add(content.file.uri.toString());
+          }
+        });
+        final String compiledBinary = await generator.recompile(invalidatedFiles);
+        if (compiledBinary != null && compiledBinary.isNotEmpty) {
+          dirtyEntries.putIfAbsent(Uri.parse(target + ".dill"),
+                  () => new DevFSFileContent(fs.file(compiledBinary)));
+          generator.accept();
```

Comment:
Thank you for highlighting this. In e9f840b I changed who instantiates IKG, so there is one copy of it per device, and I hooked up IKG accepting/rejecting logic to what comes back from VM in response to reload request.